GitUI/CommandsDialogs/SettingsDialog/Pages/BuildServerIntegrationSettingsPage.cs

```diff
@@ -26,44 +27,45 @@ protected override void Init(ISettingsPageHost aPageHost)
         {
             base.Init(aPageHost);
 
-            _populateBuildServerTypeTask =
-                Task.Factory.StartNew(() =>
+            _populateBuildServerTypeTask = ThreadHelper.JoinableTaskFactory.RunAsync(
+                async () =>
+                {
+                    await TaskScheduler.Default.SwitchTo(alwaysYield: true);
+
+                    var exports = ManagedExtensibility.GetExports<IBuildServerAdapter, IBuildServerTypeMetadata>();
```

Comment:
Ooh, this app uses MEF. I wonder if GitExtensions would benefit from a startup perf improvement from the caching the catalog as VS MEF can do.

Reply:
@drewnoakes Are you interested in verifying this?

Reply:
I haven't used MEF before. I am interested in startup perf though so can add it to the list of things to check out. Is there any urgency wrt merging this PR?

@AArnott when you say "VS MEF" are you referring to a different package, or just the way that VS uses MEF? Anything you could point me at?

Reply:
@drewnoakes I'm referring to [vs-mef](https://github.com/Microsoft/vs-mef/) which despite it's name isn't VS specific (it's just sponsored by the VS team). It's much faster than .NET MEF, and even supports caching the graph for faster startup times (if MEF is slowing down your startup at this point to begin with).